pkg/operator/hive/operatorutils.go

```diff
@@ -0,0 +1,69 @@
+package hive
+
+import (
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+	"github.com/openshift/hive/pkg/operator/assets"
+	"github.com/openshift/hive/pkg/resource"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func getHiveNamespace(config *hivev1.HiveConfig) string {
+	if config.Spec.TargetNamespace == "" {
+		return constants.DefaultHiveNamespace
+	}
+
+	return config.Spec.TargetNamespace
+}
+
+func applyAssetWithNamespaceOverride(h *resource.Helper, decoder runtime.Decoder, assetPath, namespaceOverride string) error {
+	requiredObj, err := runtime.Decode(decoder, assets.MustAsset(assetPath))
+	if err != nil {
+		return errors.Wrapf(err, "unable to decode asset: %s", assetPath)
+	}
+	obj, _ := meta.Accessor(requiredObj)
+	obj.SetNamespace(namespaceOverride)
+	_, err = h.ApplyRuntimeObject(requiredObj, scheme.Scheme)
+	if err != nil {
+		return errors.Wrapf(err, "unable to apply asset: %s", assetPath)
+	}
+	return nil
+}
+
+func applyClusterRoleBindingAssetWithSubjectNamespaceOverride(h *resource.Helper, roleBindingAssetPath, namespaceOverride string) error {
+	rb := resourceread.ReadClusterRoleBindingV1OrDie(assets.MustAsset(roleBindingAssetPath))
+	for i := range rb.Subjects {
+		if rb.Subjects[i].Kind == "ServiceAccount" || rb.Subjects[i].Namespace != "" {
+			rb.Subjects[i].Namespace = namespaceOverride
+		}
+	}
+	_, err := h.ApplyRuntimeObject(rb, scheme.Scheme)
+	if err != nil {
+		return errors.Wrapf(err, "unable to apply asset: %s", roleBindingAssetPath)
+	}
+	return nil
+}
+
+func dynamicDelete(dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, name string, hLog log.FieldLogger) error {
+	rLog := hLog.WithFields(log.Fields{
+		"gvr":  gvr,
+		"name": name,
+	})
+	if _, err := dynamicClient.Resource(gvr).Get(name, metav1.GetOptions{}); err == nil {
```

Comment:
I don't understand the purpose of doing a Get first. Why not just ignore a not-found error returned from the Delete for the case where the object does not exist? Also, the current approach is ignoring all errors from the Get, when I would think that it should only be ignoring the not-found errors.

Reply:
Good point will update.